### PR TITLE
implement callServiceExtension

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: dwds/lib/src/chrome_proxy_service.dart [hunk 1]
```diff
@@ -101,6 +101,34 @@ class ChromeProxyService implements VmServiceInterface {
     throw UnimplementedError();
   }
 
+  // TODO: Upstream this on the [VmServiceInterface].
+  Future<Response> callServiceExtension(String method,
+      {String isolateId, Map<String, String> args}) async {
+    // Validate the isolate id is correct, _getIsolate throws if not.
+    if (isolateId != null) _getIsolate(isolateId);
```
Comment (vsmenon) [inline dwds/lib/src/chrome_proxy_service.dart]: Throw if `isolateId` is null as well?

File: dwds/lib/src/chrome_proxy_service.dart [hunk 2]
```diff
@@ -101,6 +101,34 @@ class ChromeProxyService implements VmServiceInterface {
...
+    args ??= <String, String>{};
+    var expression = '''
+require("dart_sdk").developer.invokeExtension("$method", JSON.stringify(${jsonEncode(args)}));
```
Comment (vsmenon) [inline dwds/lib/src/chrome_proxy_service.dart]: Does `stringify` do the right thing if there are multiple levels of nesting?

File: dwds/lib/src/chrome_proxy_service.dart (hunk 1 above)
Comment (jakemac53) [reply dwds/lib/src/chrome_proxy_service.dart]: In the client it is treated as optional, I think that some extensions might not operate on an individual isolate? I am not totally sure though.

File: dwds/lib/src/chrome_proxy_service.dart (hunk 2 above)
Comment (jakemac53) [reply dwds/lib/src/chrome_proxy_service.dart]: This only allows `Map<String, String>`, but it does work when the value is itself a json encoded string yes (and the tests do that).

**Pull request merged**